agent/util-scripts/pbench-register-tool

```diff
@@ -222,54 +234,52 @@ fi
 
 if [[ ${remotes_arg::1} == "@" ]]; then
 	# We are dealing with a file containing the list of remotes.
+	remotes_file=${remotes_arg:1}
 
 	if [[ -n ${labels_arg} ]]; then
 		error_log "--labels=${labels_arg} not allowed with remotes file (${remotes_arg})"
 		usage >&2
 		exit 1
 	fi
-	remotes_file=${remotes_arg#*@}
-	if [[ ! -e ${remotes_file} ]]; then
-		error_log "--remotes=@${remotes_file} specifies a file that does not exist"
+
+	if [[ ! -r ${remotes_file} ]]; then
+		error_log "--remotes=@${remotes_file} specifies a file that does not exist or is not readable"
 		usage >&2
 		exit 1
 	fi
+
 	declare -a remotes_A
 	declare -A labels_A
 	idx=0
 	lineno=0
 	while read line; do
-		if [[ ${line::1} == "#" ]]; then
+		(( lineno++ ))
+		# a comment line is in regular expression form:  [ \t]*#.*\n
```

Comment:
Minor technicality, but your regex doesn't really match the code: if you don't want to use the expanded `[[:space:]]` here, then `^\s*#.*$` would be better ... (Note that `[[:blank:]]` is `[ \t]`, but `[[:space:]]` is more generous.)

Reply:
Given that the comment text is intended to complement the code, I would pull for using English instead of regex:  "A comment line is zero or more whitespace characters followed by an octothorp."  (Feel free to replace "octothorp" with a more commonly used word or phrase...  😉)

Reply:
Changed to `[:blank:]` - which is what I thought `[:space:]` was before I got disabused from that notion. I think we only want space and TAB, nothing else and certainly nothing locale-dependent.

Reply:
When I face something like this, I tend to think "so, if I'm only allowing `[:blank:]` then I should write a test to verify that other whitespace characters aren't treated as a comment" ... and then I think "why the heck would I want to do that?" and stick with `[:space:]`. 😆

Reply:
`[:space:]` includes newline (which is OK here I guess, but it may not be OK in general) and depends on locale (at least in the posix spec, which `bash(1)` says it follows, but I have not tested, nor do I plan to): that was somewhat surprising to me but things like that make me worry about unintended consequences.